kernel/kernel-api/src/main/java/io/delta/kernel/internal/replay/ActionsIterator.java

```diff
@@ -182,23 +234,51 @@ private CloseableIterator<ActionWrapper> getNextActionsIter() {
                         readSchema,
                         Optional.empty());
 
-                return combine(dataIter, false /* isFromCheckpoint */, fileVersion);
-            } else if (isCheckpointFile(nextFilePath.getName())) {
-                final long fileVersion = checkpointVersion(nextFilePath);
+                return combine(dataIter, null /* isFromCheckpoint */, fileVersion, false);
+            } else if (nextFileWrapper.isSidecar()) {
+                // The current file is a sidecar file. Because all sidecar files are appended at the
+                // end of the file list, read all sidecar files together here.
+                List<FileStatus> sidecars = Collections.singletonList(nextFile);
```

Comment:
Here we are getting a singleton list and adding more elements to  on `#245`. Does this work? I thought `Collections.singletonList` returns a unmodifiable list?

Reply:
can we add a test case where the manifest contains more than one side car file?

Reply:
Fixed (and added test).